src/wallet/wallet.cpp

```diff
@@ -3977,6 +3977,7 @@ std::vector<OutputGroup> CWallet::GroupOutputs(const std::vector<COutput>& outpu
     std::vector<OutputGroup> groups;
     std::map<CTxDestination, OutputGroup> gmap;
     CTxDestination dst;
+    std::set<CTxDestination> full_groups;
```

Comment:
ISTM a cleaner approach would be to move `groups` down before the return and just keep everything in the map. When there are more than 10 outputs in a group, just stop adding to the group. This would also handle the case of 20 outputs.

Unless our goal is to have multiple groups when more than 10 outputs for a destination?

cc @kallewoof

Reply:
The intention was that if there were a ton of spends to the same address, these would be chunked into randomized 10 UTXO chunks, to avoid creating too large transactions. So yeah, the intention was to have multiple groups for the same destination in cases where there were 11+ UTXOs for it.